OpalImagePickerDemo/OpalImagePickerDemo/MainViewController.swift

```diff
@@ -0,0 +1,145 @@
+//
+//  MainViewController.swift
+//  OpalImagePickerDemo
+//
+//  Created by Mihail Șalari on 2/9/17.
+//  Copyright © 2017 SIC. All rights reserved.
+//
+
+import UIKit
+import Photos
+import OpalImagePicker
+
+class MainViewController: UIViewController {
+
+    // MARK: - Properties
+    
+    @IBOutlet weak var collectionView: UICollectionView!
+    
+    fileprivate var images = [UIImage]()
+    
+    
+    // MARK: - LyfeCicle
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view, typically from a nib.
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+}
+
+
+// MARK: - UICollectionViewDataSource
+
+extension MainViewController: UICollectionViewDataSource {
+    
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return 1
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return images.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: MainCollectionViewCell.reuseIdentifier, for: indexPath) as! MainCollectionViewCell
+        let image = images[indexPath.row]
+        cell.imageView.image = image
+        
+        return cell
+    }
+}
+
+
+// MARK: - UICollectionViewDelegate
+
+extension MainViewController: UICollectionViewDelegate {
+    
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        let image = images[indexPath.row]
+        self.performSegue(withIdentifier: "ShowImageSegue", sender: image)
+    }
+}
+
+
+extension MainViewController {
```

Comment:
I don't like having a separate extension for each function. I like using extensions to group together implementation of a protocol, but it makes it harder to read when you declare an extension for each function. Could you group according to protocol conformance please?